components/scream/src/physics/share/scream_trcmix.cpp

```diff
@@ -0,0 +1,87 @@
+#include "scream_trcmix.hpp"
+#include "physics_constants.hpp"
+
+#include "ekat/ekat_pack_kokkos.hpp"
+
+#include <vector>
+
+using scream::Real;
+using scream::Int;
+
+extern "C" {
+
+void trcmix_c2f(const char* name, Int ncol, Int pcols, Int pver, Real* clat, Real* pmid, Real* q,
+                const Real mwdry, const Real mwco2, const Real mwn2o, const Real mwch4, const Real mwf11, const Real mwf12,
+                const Real o2mmr, const Real co2vmr_rad, const Real co2vmr, const Real n2ovmr, const Real ch4vmr, const Real f11vmr, const Real f12vmr);
+
+}
+
+namespace scream {
+namespace physics {
+
+void trcmix(
+  const char* name, Int ncol, Int pcols, Int pver,
+  trcmix_view1d<const Real> const& clat,  // latitude for columns in degrees
+  trcmix_view2d<const Real> const& pmid,  // model pressures
+  trcmix_view2d<Real>            & q,     // constituent mass mixing ratio (output)
+  const Real co2vmr_rad, const Real co2vmr, const Real n2ovmr, const Real ch4vmr, const Real f11vmr, const Real f12vmr)
+{
+  using C = Constants<Real>;
+  using P = ekat::Pack<Real, 1>;
+
+  //
+  // View bounds checking
+  //
+  EKAT_REQUIRE(clat.extent(0) == static_cast<size_t>(pcols));
+  EKAT_REQUIRE(pmid.extent(0) == static_cast<size_t>(pcols) && pmid.extent(1) == static_cast<size_t>(pver));
+  EKAT_REQUIRE(q.extent(0) == static_cast<size_t>(pcols) && q.extent(1) == static_cast<size_t>(pver));
+
+  // Transpose and copy to host. We reinterpret views as views of Pack<Real, 1> so
+  // that we can use the ekat::device_to_host, host_to_device API
+  std::vector<Real> hclat(pcols), hpmid(pcols*pver), hq(pcols*pver);
+
+  {
+    std::vector<trcmix_view1d<const Real>> stuff = {clat};
+    ekat::device_to_host(
+      {hclat.data()},
+      pcols,
+      reinterpret_cast<std::vector<trcmix_view1d<const P>>&>(stuff));
+  }
+  {
+    std::vector<trcmix_view2d<const Real>> stuff = {pmid};
```

Comment:
It seems that all you are trying to do is get a host view of the field, with sizes (pcols,pver). At this point, why not requesting a host view as input to start with? The Field class offers that possibility:
```
auto pmid = p_mid.get_view<const Real**,Host>();
```
At that point, you could roll your own nested for loops to crop `pmid` to just the physical (pcols,pver) dimensions.

I think `ekat::device_to_host` is good if you have real (pun intended) non trivial packs. Also, I suspect our impl of that fcn _assumes_ that the output host view has sizes equal to the actual allocation (that is, no padding is used).